lib/Makefile

```diff
@@ -241,6 +246,9 @@ $(ZLIB_PIC_OBJS):
 $(BZIP2_OBJS):
 	(cd $(BZIP2_DIR) && make -f Makefile-libbz2_so $(BZIP2_OBJS_BASE))
 
+$(ZSTD_OBJS):
+	$(CC) -std=c99 -Wall -Wextra -Wundef -Wshadow -Wcast-align -Wstrict-prototypes -fPIC -c $(ZSTD_DIR)/zstd.c
```

Comment:
These look like useful flags in general, can you create a CFLAGS on the file header (or even combine some with CXXFLAGS)